Make prototype submission form agency-aware

pkarman opened the pull request:
**Why**
The form should respond to the agency selected.

**How**
Refactor the request.html page to put the agency picker field first, hiding the form by default. Only show the form if the selected agency does not have an external request form already.

adborden commented on prototypes/form-wizard/_config.yml:
```diff
@@ -37,4 +37,4 @@ exclude:
   - package.json
   - README.md
-  - src
+#  - src
```
jekyll as configured doesn't know to run browserify, so this wont work. We'll need to add a watch task in npm, or a shim for jekyll to kick off `npm run build` on changes.

adborden commented on prototypes/form-wizard/src/foia.js:
```diff
@@ -1,10 +1,50 @@
-var FOIA = {
+var FOIA = {};
+
+function Agency(attributes) { Object.assign(this, attributes); }
```
Let's pull the Agency stuff into it's own file, maybe `src/models/agency.js`? Browserify will make it available by `require`:
```javascript
var Agency = require('./models/agency');
```

adborden commented on prototypes/form-wizard/src/foia.js:
```diff
@@ -1,10 +1,50 @@
 };
 
+// Agency methods
+Agency.prototype.isFOIAOnline = function () {
```
I feel like the distinction between what the technology behind the platform is not important, just the fact they have their own request form. I might just collapse the regex matching into `hasRequestForm` and we'll just add additional checks directly there. Do you think the technology distinction is important?

pkarman replied:
not really important, no. I can change to just check if `request_form` is populated.

Commit:
Make submission form agency-aware

**Why**
Agency metadata indicates, at least implicitly, which submission
methods a department supports. Respect the metadata.

**How**
Toggle the form visibility based on whether the agency selected
supports a `request_form` themselves.
